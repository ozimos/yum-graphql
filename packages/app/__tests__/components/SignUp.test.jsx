/* global React:false, shallow:false toJson:false */
import { SignUp, mapStateToProps }
  from '../../src/components/signup/ConnectedSignUp';

const props = {
  dispatch: jest.fn(),
  authenticated: false,
};

describe('SignUp Component', () => {

  const setup = () => shallow(<SignUp {...props} />);

  it('should render without throwing an error', () => {
    const wrapper = setup();

    expect((wrapper)
      .exists(<form className="form" />)).toBe(true);
  });

  it('renders correctly', () => {

    const propsAuthenticated = {
      dispatch: jest.fn(),
      authenticated: true
    };
    const propsSignUpError = {
      dispatch: jest.fn(),
      signupError: { password: 'wrong password' }
    };
    const wrapper = setup();
    expect(toJson(wrapper)).toMatchSnapshot();
    const wrapperAuthenticated = shallow(<SignUp {...propsAuthenticated} />);
    expect(toJson(wrapperAuthenticated)).toMatchSnapshot();
    const wrapperSignUpError = shallow(<SignUp {...propsSignUpError} />);
    expect(toJson(wrapperSignUpError)).toMatchSnapshot();
  });

  it('should call `componentDidUpdate` on signup error', () => {
    const serverFeedbackSpy = jest.spyOn(
      SignUp.prototype,
      'serverFeedback'
    );
    const componentDidUpdateSpy = jest.spyOn(
      SignUp.prototype,
      'componentDidUpdate'
    );
    serverFeedbackSpy.mockImplementation(() => jest.fn());
    const wrapper = setup();
    wrapper.setState({ prevSignupError: { password: 'wrong password' },
      newError: true });
    wrapper.instance().componentDidUpdate();
    expect(componentDidUpdateSpy).toHaveBeenCalled();
  });

  it('should call `handleSignupSubmit` on signup form submit', () => {
    const wrapper = setup();

    const handleSignupSubmitSpy = jest.spyOn(
      wrapper.instance(),
      'handleSignupSubmit'
    );
    wrapper.instance().handleSignupSubmit();
    expect(handleSignupSubmitSpy).toHaveBeenCalled();
  });

  it(
    'should call `disableSignupButton` on incorrect signup form input ',
    () => {
      const wrapper = setup();

      const disableSignupButtonSpy = jest.spyOn(
        wrapper.instance(),
        'disableSignupButton'
      );
      wrapper.instance().disableSignupButton();
      expect(disableSignupButtonSpy).toHaveBeenCalled();
    }
  );

  it('should call `enableSignupButton` on correct signup form input', () => {
    const wrapper = setup();

    const enableSignupButtonSpy = jest.spyOn(
      wrapper.instance(),
      'enableSignupButton'
    );
    wrapper.instance().enableSignupButton();
    expect(enableSignupButtonSpy).toHaveBeenCalled();
  });

  it('should map state to props', () => {
    const initialState = {
      loginReducer: {
        authenticated: true,
      },
      signupReducer: {
        signupError: ''
      },
    };

    const tree = mapStateToProps(initialState);
    expect(tree).toMatchSnapshot();
  });
});

