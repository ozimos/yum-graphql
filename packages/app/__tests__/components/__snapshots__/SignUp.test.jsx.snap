// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Component renders correctly 1`] = `
<div
  className="canvas signup"
>
  <div
    className="container2 container2-height"
  >
    <header
      className="header"
    >
      <h2
        className="heading"
      >
        Book A Meal
      </h2>
    </header>
    <main>
      <div
        className="welcome"
      >
        <h4
          className="welcome-text"
        >
          Welcome!
        </h4>
        <h5>
          Register by entering the information below
        </h5>
      </div>
      <div
        className="form-box"
      >
        <Formsy
          className="form"
          disabled={false}
          getErrorMessage={[Function]}
          getErrorMessages={[Function]}
          getValue={[Function]}
          hasValue={[Function]}
          isFormDisabled={[Function]}
          isFormSubmitted={[Function]}
          isPristine={[Function]}
          isRequired={[Function]}
          isValid={[Function]}
          isValidValue={[Function]}
          mapping={null}
          onChange={[Function]}
          onError={[Function]}
          onInvalid={[Function]}
          onInvalidSubmit={[Function]}
          onReset={[Function]}
          onSubmit={[Function]}
          onValid={[Function]}
          onValidSubmit={[Function]}
          preventExternalInvalidation={false}
          resetValue={[Function]}
          setValidations={[Function]}
          setValue={[Function]}
          showError={[Function]}
          showRequired={[Function]}
          validationErrors={null}
        >
          <Formsy(Input)
            innerRef={[Function]}
            name="email"
            placeholder="Email"
            required={true}
            typeOfInput="email"
            validationError="This is not a valid email"
            validationErrors={
              Object {
                "isEmail": "This is not a valid email",
                "maxLength": "input must be shorter than 50 characters",
                "minLength": "input must be longer than 5 character",
              }
            }
            validations={
              Object {
                "isEmail": true,
                "maxLength": 48,
                "minLength": 5,
              }
            }
          />
          <div
            className="invalid-feedback"
          />
          <Formsy(Input)
            innerRef={[Function]}
            name="firstName"
            placeholder="First Name"
            required={true}
            typeOfInput="text"
            validationError="Please enter your first name"
            validationErrors={
              Object {
                "maxLength": "input must be shorter than 50 characters",
                "minLength": "input must be longer than 1 character",
              }
            }
            validations={
              Object {
                "maxLength": 48,
                "minLength": 1,
              }
            }
          />
          <div
            className="invalid-feedback"
          />
          <Formsy(Input)
            innerRef={[Function]}
            name="lastName"
            placeholder="Last Name"
            required={true}
            typeOfInput="text"
            validationError="Please enter your last name"
            validationErrors={
              Object {
                "maxLength": "input must be shorter than 50 characters",
                "minLength": "input must be longer than 1 character",
              }
            }
            validations={
              Object {
                "maxLength": 48,
                "minLength": 1,
              }
            }
          />
          <Formsy(Input)
            innerRef={[Function]}
            name="password"
            placeholder="Password"
            required={true}
            typeOfInput="password"
            validationError=""
            validationErrors={
              Object {
                "hasLowerCase": "Must have lowercase letter or you can
                     use a passphrase of minimum length 16 characters",
                "hasNumber": "Must have a number or you can
                     use a passphrase of minimum length 16 characters",
                "hasSpecialCharacter": "Must have a special character or 
                    you can use a passphrase of minimum length 16 characters",
                "hasUpperCase": "Must have uppercase letter or you can 
                    use a passphrase of minimum length 16 characters",
                "maxLength": "input must be shorter than 50 characters",
                "minLength": "Must have at least 8 characters",
              }
            }
            validations={
              Object {
                "hasLowerCase": [Function],
                "hasNumber": [Function],
                "hasSpecialCharacter": [Function],
                "hasUpperCase": [Function],
                "maxLength": 48,
                "minLength": 8,
              }
            }
          />
          <Formsy(Input)
            innerRef={[Function]}
            name="confirmPassword"
            placeholder="Confirm Password"
            required={true}
            typeOfInput="password"
            validationError="Passwords do not match"
            validationErrors={
              Object {
                "equalsField": "Passwords do not match",
                "maxLength": "input must be shorter than 50 characters",
                "minLength": "Must have at least 8 characters",
              }
            }
            validations={
              Object {
                "equalsField": "password",
                "maxLength": 48,
                "minLength": 8,
              }
            }
          />
          <label
            htmlFor="isCaterer"
          >
            <Formsy(CheckBox)
              id="isCaterer"
              innerRef={[Function]}
              name="isCaterer"
              required={false}
              validationError=""
              validationErrors={Object {}}
              validations={null}
            />
            <span>
              Caterer
            </span>
          </label>
          <button
            className="btn btn-disabled"
            disabled={true}
            id="signup-button"
            type="submit"
          >
            Continue
          </button>
        </Formsy>
        <div
          className="stacked-text"
        >
          <Link
            replace={false}
            to="/login"
          >
            <p>
              Already have an account? Click here to sign in
            </p>
          </Link>
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`SignUp Component renders correctly 2`] = `
<div
  className="canvas signup"
>
  <div
    className="container2 container2-height"
  >
    <header
      className="header"
    >
      <h2
        className="heading"
      >
        Book A Meal
      </h2>
    </header>
    <main>
      <div
        className="welcome"
      >
        <h4
          className="welcome-text"
        >
          Welcome!
        </h4>
        <h5>
          Register by entering the information below
        </h5>
      </div>
      <div
        className="form-box"
      >
        <Formsy
          className="form"
          disabled={false}
          getErrorMessage={[Function]}
          getErrorMessages={[Function]}
          getValue={[Function]}
          hasValue={[Function]}
          isFormDisabled={[Function]}
          isFormSubmitted={[Function]}
          isPristine={[Function]}
          isRequired={[Function]}
          isValid={[Function]}
          isValidValue={[Function]}
          mapping={null}
          onChange={[Function]}
          onError={[Function]}
          onInvalid={[Function]}
          onInvalidSubmit={[Function]}
          onReset={[Function]}
          onSubmit={[Function]}
          onValid={[Function]}
          onValidSubmit={[Function]}
          preventExternalInvalidation={false}
          resetValue={[Function]}
          setValidations={[Function]}
          setValue={[Function]}
          showError={[Function]}
          showRequired={[Function]}
          validationErrors={null}
        >
          <Formsy(Input)
            innerRef={[Function]}
            name="email"
            placeholder="Email"
            required={true}
            typeOfInput="email"
            validationError="This is not a valid email"
            validationErrors={
              Object {
                "isEmail": "This is not a valid email",
                "maxLength": "input must be shorter than 50 characters",
                "minLength": "input must be longer than 5 character",
              }
            }
            validations={
              Object {
                "isEmail": true,
                "maxLength": 48,
                "minLength": 5,
              }
            }
          />
          <div
            className="invalid-feedback"
          />
          <Formsy(Input)
            innerRef={[Function]}
            name="firstName"
            placeholder="First Name"
            required={true}
            typeOfInput="text"
            validationError="Please enter your first name"
            validationErrors={
              Object {
                "maxLength": "input must be shorter than 50 characters",
                "minLength": "input must be longer than 1 character",
              }
            }
            validations={
              Object {
                "maxLength": 48,
                "minLength": 1,
              }
            }
          />
          <div
            className="invalid-feedback"
          />
          <Formsy(Input)
            innerRef={[Function]}
            name="lastName"
            placeholder="Last Name"
            required={true}
            typeOfInput="text"
            validationError="Please enter your last name"
            validationErrors={
              Object {
                "maxLength": "input must be shorter than 50 characters",
                "minLength": "input must be longer than 1 character",
              }
            }
            validations={
              Object {
                "maxLength": 48,
                "minLength": 1,
              }
            }
          />
          <Formsy(Input)
            innerRef={[Function]}
            name="password"
            placeholder="Password"
            required={true}
            typeOfInput="password"
            validationError=""
            validationErrors={
              Object {
                "hasLowerCase": "Must have lowercase letter or you can
                     use a passphrase of minimum length 16 characters",
                "hasNumber": "Must have a number or you can
                     use a passphrase of minimum length 16 characters",
                "hasSpecialCharacter": "Must have a special character or 
                    you can use a passphrase of minimum length 16 characters",
                "hasUpperCase": "Must have uppercase letter or you can 
                    use a passphrase of minimum length 16 characters",
                "maxLength": "input must be shorter than 50 characters",
                "minLength": "Must have at least 8 characters",
              }
            }
            validations={
              Object {
                "hasLowerCase": [Function],
                "hasNumber": [Function],
                "hasSpecialCharacter": [Function],
                "hasUpperCase": [Function],
                "maxLength": 48,
                "minLength": 8,
              }
            }
          />
          <Formsy(Input)
            innerRef={[Function]}
            name="confirmPassword"
            placeholder="Confirm Password"
            required={true}
            typeOfInput="password"
            validationError="Passwords do not match"
            validationErrors={
              Object {
                "equalsField": "Passwords do not match",
                "maxLength": "input must be shorter than 50 characters",
                "minLength": "Must have at least 8 characters",
              }
            }
            validations={
              Object {
                "equalsField": "password",
                "maxLength": 48,
                "minLength": 8,
              }
            }
          />
          <label
            htmlFor="isCaterer"
          >
            <Formsy(CheckBox)
              id="isCaterer"
              innerRef={[Function]}
              name="isCaterer"
              required={false}
              validationError=""
              validationErrors={Object {}}
              validations={null}
            />
            <span>
              Caterer
            </span>
          </label>
          <button
            className="btn btn-disabled"
            disabled={true}
            id="signup-button"
            type="submit"
          >
            Continue
          </button>
        </Formsy>
        <div
          className="stacked-text"
        >
          <Link
            replace={false}
            to="/login"
          >
            <p>
              Already have an account? Click here to sign in
            </p>
          </Link>
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`SignUp Component renders correctly 3`] = `
<div
  className="canvas signup"
>
  <div
    className="container2 container2-height"
  >
    <header
      className="header"
    >
      <h2
        className="heading"
      >
        Book A Meal
      </h2>
    </header>
    <main>
      <div
        className="welcome"
      >
        <h4
          className="welcome-text"
        >
          Welcome!
        </h4>
        <h5>
          Register by entering the information below
        </h5>
      </div>
      <div
        className="form-box"
      >
        <Formsy
          className="form"
          disabled={false}
          getErrorMessage={[Function]}
          getErrorMessages={[Function]}
          getValue={[Function]}
          hasValue={[Function]}
          isFormDisabled={[Function]}
          isFormSubmitted={[Function]}
          isPristine={[Function]}
          isRequired={[Function]}
          isValid={[Function]}
          isValidValue={[Function]}
          mapping={null}
          onChange={[Function]}
          onError={[Function]}
          onInvalid={[Function]}
          onInvalidSubmit={[Function]}
          onReset={[Function]}
          onSubmit={[Function]}
          onValid={[Function]}
          onValidSubmit={[Function]}
          preventExternalInvalidation={false}
          resetValue={[Function]}
          setValidations={[Function]}
          setValue={[Function]}
          showError={[Function]}
          showRequired={[Function]}
          validationErrors={null}
        >
          <Formsy(Input)
            innerRef={[Function]}
            name="email"
            placeholder="Email"
            required={true}
            typeOfInput="email"
            validationError="This is not a valid email"
            validationErrors={
              Object {
                "isEmail": "This is not a valid email",
                "maxLength": "input must be shorter than 50 characters",
                "minLength": "input must be longer than 5 character",
              }
            }
            validations={
              Object {
                "isEmail": true,
                "maxLength": 48,
                "minLength": 5,
              }
            }
          />
          <div
            className="invalid-feedback"
          />
          <Formsy(Input)
            innerRef={[Function]}
            name="firstName"
            placeholder="First Name"
            required={true}
            typeOfInput="text"
            validationError="Please enter your first name"
            validationErrors={
              Object {
                "maxLength": "input must be shorter than 50 characters",
                "minLength": "input must be longer than 1 character",
              }
            }
            validations={
              Object {
                "maxLength": 48,
                "minLength": 1,
              }
            }
          />
          <div
            className="invalid-feedback"
          />
          <Formsy(Input)
            innerRef={[Function]}
            name="lastName"
            placeholder="Last Name"
            required={true}
            typeOfInput="text"
            validationError="Please enter your last name"
            validationErrors={
              Object {
                "maxLength": "input must be shorter than 50 characters",
                "minLength": "input must be longer than 1 character",
              }
            }
            validations={
              Object {
                "maxLength": 48,
                "minLength": 1,
              }
            }
          />
          <Formsy(Input)
            innerRef={[Function]}
            name="password"
            placeholder="Password"
            required={true}
            typeOfInput="password"
            validationError=""
            validationErrors={
              Object {
                "hasLowerCase": "Must have lowercase letter or you can
                     use a passphrase of minimum length 16 characters",
                "hasNumber": "Must have a number or you can
                     use a passphrase of minimum length 16 characters",
                "hasSpecialCharacter": "Must have a special character or 
                    you can use a passphrase of minimum length 16 characters",
                "hasUpperCase": "Must have uppercase letter or you can 
                    use a passphrase of minimum length 16 characters",
                "maxLength": "input must be shorter than 50 characters",
                "minLength": "Must have at least 8 characters",
              }
            }
            validations={
              Object {
                "hasLowerCase": [Function],
                "hasNumber": [Function],
                "hasSpecialCharacter": [Function],
                "hasUpperCase": [Function],
                "maxLength": 48,
                "minLength": 8,
              }
            }
          />
          <Formsy(Input)
            innerRef={[Function]}
            name="confirmPassword"
            placeholder="Confirm Password"
            required={true}
            typeOfInput="password"
            validationError="Passwords do not match"
            validationErrors={
              Object {
                "equalsField": "Passwords do not match",
                "maxLength": "input must be shorter than 50 characters",
                "minLength": "Must have at least 8 characters",
              }
            }
            validations={
              Object {
                "equalsField": "password",
                "maxLength": 48,
                "minLength": 8,
              }
            }
          />
          <label
            htmlFor="isCaterer"
          >
            <Formsy(CheckBox)
              id="isCaterer"
              innerRef={[Function]}
              name="isCaterer"
              required={false}
              validationError=""
              validationErrors={Object {}}
              validations={null}
            />
            <span>
              Caterer
            </span>
          </label>
          <button
            className="btn btn-disabled"
            disabled={true}
            id="signup-button"
            type="submit"
          >
            Continue
          </button>
        </Formsy>
        <div
          className="stacked-text"
        >
          <Link
            replace={false}
            to="/login"
          >
            <p>
              Already have an account? Click here to sign in
            </p>
          </Link>
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`SignUp Component should map state to props 1`] = `
Object {
  "authenticated": true,
  "signupError": "",
}
`;
